app/controllers/api/base_controller.rb

```diff
@@ -39,8 +39,10 @@ def verify_with_otp
   end
 
   def verify_mfa_requirement
-    return if @rubygem.mfa_requirement_satisfied_for?(@api_key.user)
-    render plain: "Gem requires MFA enabled; You do not have MFA enabled yet.", status: :forbidden
+    if @rubygem && !@rubygem.mfa_requirement_satisfied_for?(@api_key.user)
+      return render plain: "Gem requires MFA enabled; You do not have MFA enabled yet.", status: :forbidden
+    end
+    render_error_if_user_mfa_requirement_unmet(@api_key.user)
   end
```

Comment:
Is there a reason why we're deciding to add to this function instead of creating a new one? I know that the rubygems metadata and user requirement both fall under the concept of "mfa requirement" and hits most of the same controllers/actions, but technically they are different features and edge cases.

I would find it confusing to have `before_action :verify_mfa_requirement, only: %i[create]` in the rubygems controller to only check the user requirement and bypass the metadata requirement to check it later because the rubygem doesn't exist.

Reply:
We started with these methods separate, but I found it really hard to wrap my head around why in some places we check certain kinds of mfa requirements and not others. It seemed to me to violate ["Tell, Don't Ask"](https://thoughtbot.com/blog/tell-dont-ask) to have all of these individual controllers needing to know exactly what kind of mfa checks to ask for, instead of keeping it in one place and letting the base controller tell the other controllers what they need.